File: Host/SlaveManager/Models/Slave.cs
using MersenneTwister;
using SharedHost.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SlaveManager.Models
{
    /// <summary>
    /// Slavedevice is the cloud computer device that allow user to access 
    /// </summary>
    public class Slave
    {
        public Slave()
        { }

        public Slave(SlaveDeviceInformation information)
        {
            CPU = information.CPU;
            GPU = information.GPU;
            RAMcapacity = information.RAMcapacity;
            OS = information.OS;
            ID = information.ID;
        }


        /// <summary>
        /// Each slave device defined with an unique ID 
        /// </summary>
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.None)]
        public int ID { get; set; }


        public DateTime? Register { get; set; }


        /// <summary>
        /// slave device hardware configuration
        /// </summary>
        public string CPU { get; set; }
        public string GPU { get; set; }
        public int RAMcapacity { get; set; }
        public string OS { get; set; }

        /// <summary>
        /// 
        /// </summary>
        public virtual ICollection<CommandLog> CommandLogs { get; set; }

        /// <summary>
        /// (nullable) if slave is in a session, 
        /// </summary>
        public virtual ICollection<Session> servingSession { get; set; }

        /// <summary>
        /// 
        /// </summary>
        public virtual ICollection<GeneralError> GeneralErrors { get; set; }
        
        /// <summary>
        /// 
        /// </summary>
        public virtual ICollection<SessionCoreExit> SessionCoreExits { get; set; }
    }
}

File: Client/admin_client/lib/models/Slave.dart
class Slave {
  final int ID;
  final String CPU;
  final String GPU;
  final int RAMcapacity;
  final String OS;
  final String CommandLogs;
  final String ServingSession;
  final String GeneralErrors;
  final String SessionCoreExits;

  final String Register;

  Slave({
    required this.ID,
    required this.CPU,
    required this.GPU,
    required this.RAMcapacity,
    required this.OS,
    required this.CommandLogs,
    required this.ServingSession,
    required this.GeneralErrors,
    required this.SessionCoreExits,
    required this.Register,
  });

  factory Slave.fromJson(Map<String, dynamic> json_outer) {
    var json = json_outer['Item1'];
    print(json['servingSession']);

    return Slave(
      ID: json['ID'],
      CPU: json['CPU'],
      GPU: json['GPU'],
      RAMcapacity: json['RAMcapacity'],
      OS: json['OS'],
      CommandLogs: "json['CommandLogs']" as String,
      ServingSession: "json['servingSession']" as String,
      GeneralErrors: "json['GeneralErrors']" as String,
      SessionCoreExits: "json['SessionCoreExits']" as String,
      Register: json['Register'] as String,
    );
  }
}
